### ISSUE TITLE
Many summarizer sessions never start summary process

### ISSUE THREAD

**Issue by vladsud** [microsoft/FluidFramework#9884]
Started to look into this area based on observations from our failing stress tests.

Here is an interesting query - it looks at all summary sessions that are longer than 10 minutes:

**let times** = union Office_Fluid_FluidRuntime_*
| where Data_clientType != "interactive"
| summarize low=min(Event_Time), high=max(Event_Time) by Data_containerId
| extend len = totimespan(high-low) / 1s
| where len > 600;
**let sessions** = union Office_Fluid_FluidRuntime_*
| where Data_clientType != "interactive"
| where Data_eventName contains "Container:Load_end" or Data_event contains "CreatingSummarizer" or Data_eventName contains "RunningSummarizer" or Data_eventName contains "ConnectionStateChange_Connected"
| summarize count() by Data_eventName, Data_containerId;
**times | join (sessions)** on $left.Data_containerId == $right.Data_containerId
| summarize count() by Data_eventName

| event | count |
|-|-|
| fluid:telemetry:Container:Load_end | 169,570 |
| fluid:telemetry:Container:ConnectionStateChange_Connected | 169,368 |
| fluid:telemetry:Summarizer:RunningSummarizer | 162,907 |

You see that over 6K sessions never moved to doing any summaries in 10 minutes!

If you change the interval to 60 seconds vs. 600 seconds, you will learn that only 1.4K sessions did not connect in 1 minute.
This means that across 600 second sessions, there were even fewer than 1.3K sessions that did not connect in 1 minute. We can assume that at there are least 5K sessions that were longer than 10 minutes, connected in less than a minute, but did not start summarizing in 10 minutes! We can also eliminate long boot case - there were only 333 sessions that tool longer than a minute to load container.

We can also look at all sessions, and how many of them moved to running summarizer (compared to number of loaded sessions):

| event | count | % of sessions |
|-|-|-|
| fluid:telemetry:Container:Load_end | 3,728,732 | 100%| 
| fluid:telemetry:Container:ConnectionStateChange_Connected | 3,711,380 | 99.5% |
| fluid:telemetry:Summarizer:RunningSummarizer | 3,640,419 | 97.6% |

Some of these were closed too soon, so it's hard to make any judgement here (without time-based analyses as above), but nevertheless it shows we have 2% of summarizer sessions that never start summarization.

**Comment by vladsud** [comment]
Please note that we can bail out soon after waiting, and before reaching fluid:telemetry:Summarizer:RunningSummarizer event. But if we do, summarizer session should end right away, it should not last 10 minutes!

```ts
            // Re-validate that it need to be running. Due to asynchrony, it may be not the case anymore
            // but only if creation was delayed. If it was not, then we want to ensure we always create
            // a summarizer to kick off lastSummary. Without that, we would not be able to summarize and get
            // document out of broken state if it has too many ops and ordering service keeps nacking main
            // container (and thus it goes into cycle of reconnects)
            if (startWithInitialDelay && this.getShouldSummarizeState().shouldSummarize === false) {
                return;
            }
```

**Comment by vladsud** [comment]
Here is what I see based on test automation:

> union office_fluid_ffautomation_*
| where Data_docId == "ZB5RoS43AXy%2FVSoSAOKiWwXkuUBG6lVcJcjv4xubag4%3D"
| where Data_loaderId == "2ca86e77-89e9-450b-8e7b-7a38d9cb273a"
| where Event_Sequence > 357
| project Event_Sequence, Event_Time, Data_eventName, Data_reason, Data_clientType, Data_containerId

Main was closed and bailed out while summarizer client was booting.
> 365	2022-04-08T22:40:31.713Z	fluid:telemetry:Container:ContainerClose		interactive	8b9143ef-d22d-4eee-93cf-63ae3b1a5ffe

Summarizer connected just fine, and actually noticed that it is not a summarizer:

> 383	2022-04-08T22:40:31.744Z	fluid:telemetry:SummaryManager:EndingSummarizer	unknown	interactive	8b9143ef-d22d-4eee-93cf-63ae3b1a5ffe

But it did absolutely nothing for next 10 minutes!

And I think with https://github.com/microsoft/FluidFramework/pull/8951, this will mean that no other summarizer will start as they will consider this client to be a summarizer.

In other words, we have a bug, and this bug caused summarizer clients (in some cases) to stick around longer than they should. But because other clients would not consider them as summarizer anymore, there was no visible impact to the system.
But with a recent change it made this bug result in substantial impact to the system.

That's just a theory based on observing one session, and data from prod.

**Comment by vladsud** [comment]
Given that I do not see "StoppingSummarizer" event in telemetry, I assume we never got to Summarizer.runCore() method.

This likely points to a fact that startSummarization() existed here:
```ts
            // Re-validate that it need to be running. Due to asynchrony, it may be not the case anymore
            const shouldSummarizeState = this.getShouldSummarizeState();
            if (shouldSummarizeState.shouldSummarize === false) {
                this.state = SummaryManagerState.Starting;
                summarizer.stop(shouldSummarizeState.stopReason);
                return;
            }
```

Summarizer.stop() does not do match. What we are missing is these steps that are running in most cases to close summarizer container session:

```
    public async run(
        onBehalfOf: string,
        options?: Readonly<Partial<ISummarizerOptions>>): Promise<SummarizerStopReason> {
        try {
            return await this.runCore(onBehalfOf, options);
        } catch (error) {
            this.stop("summarizerException");
            throw SummarizingWarning.wrap(error, false /* logged */, this.logger);
        } finally {
            this.dispose();
            this.runtime.closeFn();
        }
    }
```

Notice dispose() & closeFn() are not called anywhere else!
Nit: I think we should not call dispose() - that should happen as part of close flow.
